### ISSUE TITLE
Gradle console output breaks pinentry

### ISSUE THREAD

**Issue by britter** [gradle/gradle#11706]
When using pinentry to provide password for gpg keys to gpg-agent, Gradles console output breaks pinentry by sending output to the console.

### Expected Behavior

Gradle should stop sending output while gpg and pinentry are running

### Current Behavior

Gradle's output breaks pinentry making in impossible to provide the password

Using Gradle's rich console it looks like this:

<img width="1130" alt="Bildschirmfoto 2019-12-15 um 14 52 55" src="https://user-images.githubusercontent.com/1327662/70994988-b67a2a00-20cf-11ea-9ccb-668e8cf6df95.png">

Pressing tab to navigate to the cancel button adds more output to the pinentry window:

<img width="1130" alt="Bildschirmfoto 2019-12-15 um 14 53 18" src="https://user-images.githubusercontent.com/1327662/70995048-de698d80-20cf-11ea-9c68-6cc28e6c07c5.png">

Starting Gradle with `--console=plain` also doesn't work. It feels like pinentry only receives very second keystroke and sometimes Gradle's output will also be added to pinentry:

<img width="1130" alt="Bildschirmfoto 2019-12-17 um 13 13 25" src="https://user-images.githubusercontent.com/1327662/70995131-0fe25900-20d0-11ea-94e3-1250d19e902f.png">

A workaround is to sign some abitrary file to force gpg-agent to store the password for the key and then run the build directly after that.

### Context

I'm trying to setup up a Gradle build for one of my side projects and I want to use gpg-agent to pass the password for my gpg key.

### Steps to Reproduce 

* Setup gpg with gpg-agent on your machine and use pinentry to supply the password.
* Setup a project for signing with `useGpgCmd()`
* Run a build with signing

### Your Environment

_I can't supply a build scan because once pinentry starts I have to terminate the build_

* macOS Catalina 10.15.1
* fish shell version 3.0.2
* gpg (GnuPG) 2.2.19
* `cat ~/.gnupg/gpg-agent.conf` -> `pinentry-program /usr/local/bin/pinentry-curses`

**Comment by lacasseio** [comment]
I don't think this is much of a bug as it was never really supported. Did it ever worked in previous Gradle version?

**Comment by britter** [comment]
I don't think this worked for the time we have the rich console now. I may have worked before but I don't know. Would you like me to rephrase this as a feature request?

**Comment by britter** [comment]
@gradle/core I think this is still valid.

**Comment by britter** [comment]
@gradle/bt-core I think this is still valid.
